appcues/src/main/java/com/appcues/ui/composables/AppcuesComposition.kt

```diff
@@ -91,44 +95,63 @@ private fun BoxScope.ComposeLastRenderingState(state: Rendering) {
     }
 
     with(state.stepContainer) {
-        // apply backdrop traits
-        ApplyBackgroundDecoratingTraits(list = backdropDecoratingTraits.toMutableList())
-
-        // create wrapper
-        contentWrappingTrait.WrapContent { hasFixedHeight, contentPadding ->
-            Box(contentAlignment = Alignment.TopCenter) {
-                ApplyUnderlayContainerTraits(this)
-
-                // Apply content holder trait
-                with(contentHolderTrait) {
-                    // create object that will passed down to CreateContentHolder
-                    ContainerPages(
-                        pageCount = steps.size,
-                        currentPage = state.position,
-                        composePage = { index ->
-                            with(steps[index]) {
-                                CompositionLocalProvider(LocalAppcuesActions provides actions) {
-                                    // used to get the padding values from step decorating trait and apply to the Column
-                                    val density = LocalDensity.current
-                                    val stepDecoratingPadding = remember(this) { StepDecoratingPadding(density) }
-
-                                    ApplyUnderlayStepTraits(this@Box, stepDecoratingPadding)
-
-                                    ComposeStepContent(index, hasFixedHeight, contentPadding, stepDecoratingPadding)
-
-                                    ApplyOverlayStepTraits(this@Box, stepDecoratingPadding)
+        val backdropDecoratingTraits = remember(state.position) { mutableStateOf(steps[state.position].backdropDecoratingTraits) }
+        val containerDecoratingTraits = remember(state.position) { mutableStateOf(steps[state.position].containerDecoratingTraits) }
+        val metadataSettingTraits = remember(state.position) { mutableStateOf(steps[state.position].metadataSettingTraits) }
+        val previousStepMetaData = remember { mutableStateOf(AppcuesStepMetadata()) }
+        val stepMetadata = remember(metadataSettingTraits.value) {
```

Comment:
will try to refactor this into a simpler method, but its basically observing changes from metadataSettingsTraits and when that happens its iterating over all those traits and producing a new set of stepMetadata, used by LocalAppcuesStepMetadata below